test/jdk/sun/net/www/http/HttpClient/KeepAliveTest.java

```diff
@@ -0,0 +1,590 @@
+     */
+    private volatile boolean isProxySet = false;
+    private volatile boolean serverReady = false;
+    private static final String NI = "NO_INPUT";
+    private static final String CONNECTION_KEEP_ALIVE_ONLY = "Connection: keep-alive";
+    private static final String PROXY_CONNECTION_KEEP_ALIVE_ONLY = "Proxy-Connection: keep-alive";
+    private static final String KEEP_ALIVE_TIMEOUT_NEG = "Keep-alive: timeout=-20";
+    private static final String KEEP_ALIVE_TIMEOUT_ZERO = "Keep-alive: timeout=0";
+    private static final String KEEP_ALIVE_TIMEOUT = "Keep-alive: timeout=20";
+    private static final String KEEP_ALIVE_PROXY_TIMEOUT = "Keep-alive: timeout=120";
+    private static final String CLIENT_HTTP_KEEP_ALIVE_TIME_SERVER_NEGATIVE = "http.keepAlive.time.server=-100";
+    private static final String CLIENT_HTTP_KEEP_ALIVE_TIME_PROXY_NEGATIVE = "http.keepAlive.time.proxy=-200";
+    private static final String CLIENT_HTTP_KEEP_ALIVE_TIME_SERVER_ZERO = "http.keepAlive.time.server=0";
+    private static final String CLIENT_HTTP_KEEP_ALIVE_TIME_PROXY_ZERO = "http.keepAlive.time.proxy=0";
+    private static final String CLIENT_HTTP_KEEP_ALIVE_TIME_SERVER_POSITIVE = "http.keepAlive.time.server=100";
+    private static final String CLIENT_HTTP_KEEP_ALIVE_TIME_PROXY_POSITIVE = "http.keepAlive.time.proxy=200";
+    private static final String CONNECTION_KEEP_ALIVE_WITH_TIMEOUT = CONNECTION_KEEP_ALIVE_ONLY + NEW_LINE
+        + KEEP_ALIVE_TIMEOUT;
+    private static final String SERVER_100_NEGATIVE = CLIENT_HTTP_KEEP_ALIVE_TIME_SERVER_NEGATIVE;
+    private static final String PROXY_200_NEGATIVE = CLIENT_HTTP_KEEP_ALIVE_TIME_PROXY_NEGATIVE;
+    private static final String SERVER_ZERO = CLIENT_HTTP_KEEP_ALIVE_TIME_SERVER_ZERO;
+    private static final String PROXY_ZERO = CLIENT_HTTP_KEEP_ALIVE_TIME_PROXY_ZERO;
+    private static final String SERVER_100 = CLIENT_HTTP_KEEP_ALIVE_TIME_SERVER_POSITIVE;
+    private static final String PROXY_200 = CLIENT_HTTP_KEEP_ALIVE_TIME_PROXY_POSITIVE;
+    private static final String A = CONNECTION_KEEP_ALIVE_ONLY;
+    private static final String B = CONNECTION_KEEP_ALIVE_WITH_TIMEOUT;
+    private static final String C = PROXY_CONNECTION_KEEP_ALIVE_ONLY;
+    private static final String D = PROXY_CONNECTION_KEEP_ALIVE_ONLY + NEW_LINE + CONNECTION_KEEP_ALIVE_ONLY;
+    private static final String E = C + NEW_LINE + KEEP_ALIVE_PROXY_TIMEOUT;
+    private static final String F = A + NEW_LINE + KEEP_ALIVE_TIMEOUT_NEG;
+    private static final String G = A + NEW_LINE + KEEP_ALIVE_TIMEOUT_ZERO;
+    private static final String H = C + NEW_LINE + KEEP_ALIVE_TIMEOUT_NEG;
+    private static final String I = C + NEW_LINE + KEEP_ALIVE_TIMEOUT_ZERO;
+    private static Logger logger = Logger.getLogger("sun.net.www.protocol.http.HttpURLConnection");
+    private String[] serverScenarios = {
```

Comment:
should be final

Reply:
now serverScenarios is retrieved using getServerScenarion(int) and serverScenarios array is commented.